File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {

        Set<String> res= new HashSet<>();
        Set<String> set= new HashSet<>(); 
//  0
// 10 
// 1, 11
// 2, 12


        for(int i=0;i<=s.length()-10; i++){
            String temp= s.substring(i,i+10) ;
            if(set.contains(temp) ) 
                res.add(temp); 

            set.add(temp); 
            

        }
        return new ArrayList<>(res); 
    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        res= set()
        seen = set()

        for i in range(0,len(s)-9):
            temp = s[i:i+10]
            if temp in seen: 
                res.add(temp)
            seen.add(temp)
        
        return list(res)
        